subiquity/common/filesystem/gaps.py

```diff
@@ -230,3 +230,14 @@ def at_offset(device, offset):
             if pg.offset == offset:
                 return pg
     return None
+
+
+def intersects(device, gap):
+    """Find the first gap that is a subset of the supplied gap"""
```

Comment:
I have a naming quibble here. intersect to me returns a gap that has overlap with the supplied gap, not one that's contained in it. Maybe "def gap_within" or something?

Reply:
Went with gaps.within, since most things access these methods as gaps.X.
See new commits.